dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/ModelChecker.java

```diff
@@ -80,11 +86,21 @@ public static void preprocessMemoryModel(VerificationTask task, Configuration co
      * @exception UnsatisfiedRequirementException Some static analysis is missing.
      */
     public static void performStaticProgramAnalyses(VerificationTask task, Context analysisContext, Configuration config) throws InvalidConfigurationException {
-        Program program = task.getProgram();
+        final Program program = task.getProgram();
         analysisContext.register(BranchEquivalence.class, BranchEquivalence.fromConfig(program, config));
-        analysisContext.register(ExecutionAnalysis.class, ExecutionAnalysis.fromConfig(program, task.getProgressModel(), analysisContext, config));
-        analysisContext.register(ReachingDefinitionsAnalysis.class, ReachingDefinitionsAnalysis.fromConfig(program, analysisContext, config));
-        analysisContext.register(AliasAnalysis.class, AliasAnalysis.fromConfig(program, analysisContext, config));
+        analysisContext.register(ExecutionAnalysis.class, ExecutionAnalysis.fromConfig(program, task.getProgressModel(),
+                analysisContext, config));
+        analysisContext.register(ReachingDefinitionsAnalysis.class, ReachingDefinitionsAnalysis.fromConfig(program,
+                analysisContext, config));
+        final AliasAnalysis alias = AliasAnalysis.fromConfig(program, analysisContext, config, logger.isWarnEnabled());
```

Comment:
AFAICS, this parameter control functionality (e.g., it might enable tearing) and as such, it should not be decided on a log level

Reply:
The MSA detection there is only done for the purpose of logging, so it is unnecessary when you don't want to log it. Whether this optimization is necessary or not is a different story.

Note that `Tearing` runs before this piece of code using its own instance of AA (we run AA twice now if MSA is enabled, once for tearing, once at the end).

Reply:
Are you saying that we run the analysis again *just* to log if there are potential MSA? Why can't we do this the first time we run the analysis?

Reply:
The first time AA runs only if MSA Detection + Tearing is enabled, meaning if MSA is detected it also gets removed.
Since the program gets changed by Tearing, the results of AA get invalidated and that is why we need to run it a second time. However, the second time we run the MSA detection part of it only for the purpose of logging. The rest of the alias analysis is still necessary.

In short: The alias analysis in `ModelChecker` always needs to run, but its MSA detection part is only for logging. Only if MSA Detection/Tearing is enabled, there is another run of AA that preceeds the one in `ModelChecker`.